src/Agent.Worker/Handlers/Handler.cs

```diff
@@ -138,6 +138,13 @@ protected void AddVariablesToEnvironment(bool excludeNames = false, bool exclude
                 string formattedKey = (pair.Key ?? string.Empty).Replace('.', '_').ToUpperInvariant();
                 AddEnvironmentVariable(formattedKey, pair.Value);
 
+                // for maintaining back compat variable names replace space with underscore
+                string backCompatKey = formattedKey.Replace(' ', '_');
```

Comment:
IMO, we should only be replacing dot with underscore. or atleast thats what we should say. Replacing space with underscore is done only because the old agent was doing.